File: Testing/Code for Testing/Early Basic Syntax Testing/result/test_result.js

function divide(a, b) {
    if ((b === 0)) {
        return {ok: false, error: "division by zero"};
    }
    return {ok: true, value: Math.trunc(a / b)};
}

function safe_calc(x, y) {
    const _r0 = divide(x, y);
    if (!_r0.ok) {
        return _r0;
    }
    let result = _r0.value;
    return {ok: true, value: (result * 2)};
}

function main() {
    const _r1 = divide(10, 2);
    if (!_r1.ok) {
        console.error("error: " + _r1.error);
        process.exit(1);
    }
    let a = _r1.value;
    console.log(a);
    const _r2 = safe_calc(20, 4);
    if (!_r2.ok) {
        console.error("error: " + _r2.error);
        process.exit(1);
    }
    let b = _r2.value;
    console.log(b);
    const _r3 = divide(5, 0);
    if (!_r3.ok) {
        console.error("error: " + _r3.error);
        process.exit(1);
    }
    let c = _r3.value;
    console.log(c);
    return 0;
}

function __ul_sort(arr, n) { const sub = arr.slice(0, n).sort((a, b) => a < b ? -1 : a > b ? 1 : 0); for (let i = 0; i < n; i++) arr[i] = sub[i]; }

main();

File: Testing/Code for Testing/Early Basic Syntax Testing/result/test_result.py
from dataclasses import dataclass, field
import sys

def _ul_sort(arr, n): arr[:n] = sorted(arr[:n])

class _ul_Exception(Exception):
    def __init__(self, type_name, message):
        self.type_name = type_name
        self.message = message
        super().__init__(message)


def divide(a, b):
    if (b == 0):
        return (False, 'division by zero')
    return (True, (a // b))

def safe_calc(x, y):
    _r0 = divide(x, y)
    if not _r0[0]:
        return _r0
    result = _r0[1]
    return (True, (result * 2))

def main():
    _r1 = divide(10, 2)
    if not _r1[0]:
        print("error: " + str(_r1[1]), file=sys.stderr)
        sys.exit(1)
    a = _r1[1]
    print(a)
    _r2 = safe_calc(20, 4)
    if not _r2[0]:
        print("error: " + str(_r2[1]), file=sys.stderr)
        sys.exit(1)
    b = _r2[1]
    print(b)
    _r3 = divide(5, 0)
    if not _r3[0]:
        print("error: " + str(_r3[1]), file=sys.stderr)
        sys.exit(1)
    c = _r3[1]
    print(c)
    return 0


if __name__ == "__main__":
    main()
